PR: https://github.com/odoo/odoo/pull/

From: 40fa74e3f561327979d8b56e23e7630291d26683
From: Gaurav Panchal
Date: 2016-06-30 12:41:21

Structural Changes: 9
Total Changes: 100

[MIG] project: migration to new api except project.py

================================= pseudo patch: =================================

--- a/addons/project/models/res_company.py
+++ b/addons/project/models/res_company.py
@@ -1,15 +1,13 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from odoo import fields, models
 
-class res_company(osv.osv):
+
+class ResCompany(models.Model):
     _inherit = 'res.company'
-    _columns = {
-        'project_time_mode_id': fields.many2one('product.uom', 'Project Time Unit',
-            help='This will set the unit of measure used in projects and tasks.\n' \
-"If you use the timesheet linked to projects, don't " \
-"forget to setup the right unit of measure in your employees.",
-        ),
-    }
+
+    project_time_mode_id = fields.Many2one('product.uom', string='Project Time Unit',
+        help="This will set the unit of measure used in projects and tasks.\n"
+             "If you use the timesheet linked to projects, don't "
+             "forget to setup the right unit of measure in your employees.")

--- a/addons/project/models/res_config.py
+++ b/addons/project/models/res_config.py
@@ -1,36 +1,35 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from openerp import SUPERUSER_ID
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from odoo import api, fields, models
 
-class project_configuration(osv.osv_memory):
+
+class ProjectConfiguration(models.TransientModel):
     _name = 'project.config.settings'
     _inherit = 'res.config.settings'
 
-    _columns = {
-        'module_pad': fields.selection([
-            (0, "Task description is a plain text"),
-            (1, "Collaborative rich text on task description")
-            ], "Pads",
-            help='Lets the company customize which Pad installation should be used to link to new pads '
-                 '(for example: http://ietherpad.com/).\n'
-                 '-This installs the module pad.'),
-        'module_rating_project': fields.selection([
-            (0, "No customer rating"),
-            (1, 'Allow activating customer rating on projects, at issue completion')
-            ], "Rating",
-            help="This allows customers to give rating on provided services"),
-        'generate_project_alias': fields.selection([
-            (0, "Do not create an email alias automatically"),
-            (1, "Automatically generate an email alias at the project creation")
-            ], "Project Alias",
-            help="Odoo will generate an email alias at the project creation from project name."),
-        'module_project_forecast': fields.boolean("Forecasts, planning and Gantt charts"),
-    }
+    module_pad = fields.Selection([
+        (0, "Task description is a plain text"),
+        (1, "Collaborative rich text on task description")
+        ], string="Pads",
+        help='Lets the company customize which Pad installation should be used to link to new pads '
+             '(for example: http://ietherpad.com/).\n'
+             '-This installs the module pad.')
+    module_rating_project = fields.Selection([
+        (0, "No customer rating"),
+        (1, 'Allow activating customer rating on projects, at issue completion')
+        ], string="Rating",
+        help="This allows customers to give rating on provided services")
+    generate_project_alias = fields.Selection([
+        (0, "Do not create an email alias automatically"),
+        (1, "Automatically generate an email alias at the project creation")
+        ], string="Project Alias",
+        help="Odoo will generate an email alias at the project creation from project name.")
+    module_project_forecast = fields.Boolean(string="Forecasts, planning and Gantt charts")
 
-    def set_default_generate_project_alias(self, cr, uid, ids, context=None):
-        config_value = self.browse(cr, uid, ids, context=context).generate_project_alias
-        check = self.pool['res.users'].has_group(cr, uid, 'base.group_system')
-        self.pool.get('ir.values').set_default(cr, check and SUPERUSER_ID or uid, 'project.config.settings', 'generate_project_alias', config_value)
+    @api.multi
+    def set_default_generate_project_alias(self):
+        check = self.env.user.has_group('base.group_system')
+        Values = check and self.env['ir.values'].sudo() or self.env['ir.values']
+        for config in self:
+            Values.set_default('project.config.settings', 'generate_project_alias', config.generate_project_alias)

--- a/addons/project/models/res_partner.py
+++ b/addons/project/models/res_partner.py
@@ -1,19 +1,16 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from openerp.osv import fields,osv
+from odoo import fields, models
 
-class res_partner(osv.osv):
-    def _task_count(self, cr, uid, ids, field_name, arg, context=None):
-        Task = self.pool['project.task']
-        return {
-            partner_id: Task.search_count(cr,uid, [('partner_id', '=', partner_id)], context=context)
-            for partner_id in ids
-        }
-    
+
+class ResPartner(models.Model):
     """ Inherits partner and adds Tasks information in the partner form """
     _inherit = 'res.partner'
-    _columns = {
-        'task_ids': fields.one2many('project.task', 'partner_id', 'Tasks'),
-        'task_count': fields.function(_task_count, string='# Tasks', type='integer'),
-    }
+
+    task_ids = fields.One2many('project.task', 'partner_id', string='Tasks')
+    task_count = fields.Integer(compute='_compute_task_count', string='# Tasks')
+
+    def _compute_task_count(self):
+        for partner in self:
+            partner.task_count = len(partner.task_ids)

--- a/addons/project/models/web_planner.py
+++ b/addons/project/models/web_planner.py
@@ -1,5 +1,7 @@
 # -*- coding: utf-8 -*-
-from openerp import models
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
 
 
 class PlannerProject(models.Model):
